### PR TITLE
Switch to esbuild and tsc instead of pika

### PR DESCRIPTION
**Pull request by kfcampbell**
Replicating [octokit/request.js#584](https://github.com/octokit/request.js/pull/584) to another repo. 

---

Pika has been deprecated for a while now, and the project has now been archived, and because of that the released npm package is missing most of the files generated by the build step.

Uses esbuild to transpile the TS source code into an ESM source, NodeJs bundle, and a browser bundle

Uses the TypeScript compiler to generate the types.

### Before the change?

* Uses pika for the build system
* The released npm package is missing most of the files generated by the build step. `dist-node`, `dist-types`, `dist-web`... even though they are generated correctly.

### After the change?

* Uses esbuild to generate the JS bundles
* Uses the TypeScript compiler to generate the type definitions
* The correct file matching patterns are used so we publish all the necessary files again.

----

### Pull request checklist
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Added the appropriate label for the given change

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/main/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No

If `Yes`, what's the impact:

* N/A

### Pull request type

Please add the corresponding label for change this PR introduces:
- Bugfix: `Type: Bug`
- Dependencies/code cleanup: `Type: Maintenance`

----

### CODE REVIEW

**Comment by wolfy1339** [comment]
Don't forget to explicitly mark type imports and exports.

Whenever an imported value, that is also JS code, is only used as a type, it should be imported as a type, to make sure that it doesn't show up in the build output.

All typescript types (except for `enum`) need to be imported using type imports, and exports need to use the type export syntax.

**Comment by kfcampbell** [comment]
@wolfy1339 I've updated this PR to correctly import types, bumped to the latest version of octokit/tsconfig, and fixed the merge conflict! Hopefully this is good to go now.

**Comment by kfcampbell** [comment]
Hmm...locally I'm seeing:

image

and:

image

Even after many reloads here, I'm not seeing the new commits: 

image

I wonder if we're having issues.

**Comment by kfcampbell** [comment]
Yep, looks like that's it:

image

**Comment by kfcampbell** [comment]
Looks like we're all good now!

File: scripts/build.mjs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  const entryPoints = ["./pkg/dist-src/index.js"];
+
+  await Promise.all([
+    // Build the a CJS Node.js bundle
+    esbuild.build({
+      entryPoints,
+      outdir: "pkg/dist-node",
+      bundle: true,
+      platform: "node",
+      target: "node14",
```
[the remark this answers is not in the document]
Comment (kfcampbell) [reply scripts/build.mjs:45]: Ahh, thanks! I missed that the build script also needed to be updated for the Node target.

**Review by wolfy1339 — DISMISSED** [review]

**Review by wolfy1339 — APPROVED** [review]

**Pull request merged**